quill-jdbc/src/main/scala/io/getquill/sources/jdbc/JdbcSource.scala

```diff
@@ -51,68 +54,73 @@ class JdbcSource[D <: SqlIdiom, N <: NamingStrategy](config: JdbcSourceConfig[D,
       }
     }
 
-  def transaction[T](f: => T) =
-    withConnection { conn =>
-      currentConnection.withValue(Some(conn)) {
-        conn.setAutoCommit(false)
-        try {
-          val res = f
-          conn.commit
-          res
-        } catch {
-          case NonFatal(e) =>
-            conn.rollback
-            throw e
-        } finally {
-          conn.setAutoCommit(true)
-        }
+  def transaction[T](f: JdbcSource[D, N] => T) = {
```

Comment:
Same suggestion, this one works but it will create a new connection pool for each transaction.